Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -236,6 +238,52 @@ lemma odd_matches_node_outside {u : Set V} {c : ConnectedComponent (Subgraph.del
   exact hMeven
 
 end Finite
+
+lemma isClique_even_iff_exists_isMatching [DecidableEq V] (u : Set V) (hu : Set.Finite u)
+    (hc : G.IsClique u) : Even (Nat.card u) ↔ ∃ (M : Subgraph G), M.verts = u ∧ M.IsMatching := by
```

Comment:
This result is still true if `u` is infinite: `Nat.card u = 0` is even and `u` contains a matching on `u`.
```suggestion
lemma isClique_even_iff_exists_isMatching [DecidableEq V] (u : Set V) (hc : G.IsClique u) :
    Even (Nat.card u) ↔ ∃ (M : Subgraph G), M.verts = u ∧ M.IsMatching := by
```

Reply:
I've adapted it to make room for it, but I'm not sure how to get there. The proof strategy for the finite case doesn't work, so I went looking for some sort of lemma about partitioning an infinite set into two infinite sets, but I can't find it. Any thoughts on how to approach this part?

Reply:
With some help from @chrisflav and @Raph-DG I have a rough idea how to get there through `Cardinal`. I'll work that out soonish:)